### PR TITLE
Implement ci test

### LINKED ISSUE #41: Implement a continuous integration test with github actions
### Description of feature

It could be something similar to this:
https://github.com/nf-core/taxprofiler/blob/dev/.github/workflows/ci.yml

### PR DESCRIPTION
**Pull request by ljmesi**
Fixes #41

### CODE REVIEW

File: assets/ci/README.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+-t 92 \
+"${INDEX}" \
+"${FASTQ_FILES_DIR}"/pe/SRR12875570_1.fastq.gz "${FASTQ_FILES_DIR}"/pe/SRR12875570_2.fastq.gz > "${TAXID}"_pe.bam) \
+&> "${TAXID}"_bwa_mem.log
+(samtools sort \
+-@ 47 \
+-n \
+"${TAXID}"_pe.bam \
+| samtools view \
+-f 4 \
```
Comment (sofstam) [inline assets/ci/README.md]: Should this be `-F 4`?

File: assets/ci/README.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+INDEX=`find -L ./ -name "*.amb" | sed 's/.amb//'`
+(bwa mem \
+-v 3 \
+-t 92 \
+"${INDEX}" \
+"${FASTQ_FILES_DIR}"/pe/SRR12875570_1.fastq.gz "${FASTQ_FILES_DIR}"/pe/SRR12875570_2.fastq.gz > "${TAXID}"_pe.bam) \
+&> "${TAXID}"_bwa_mem.log
+(samtools sort \
+-@ 47 \
```
Comment (sofstam) [inline assets/ci/README.md]: If this is the number of threads maybe reduce it?

File: assets/ci/README.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+(minimap2 \
+-t 92 \
+"${TAXID}".mmi \
+"${FASTQ_FILES_DIR}"/se/SRR12875558_1.fastq.gz \
+-L \
+-a \
+| samtools sort \
+-@ 47 \
+| samtools view -@ 47 \
+--require-flags 4 \
```
Comment (sofstam) [inline assets/ci/README.md]: `-F 4 `also here?

File: assets/ci/README.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+&> minimap2_index.log
+
+(minimap2 \
+-t 92 \
+"${TAXID}".mmi \
+"${FASTQ_FILES_DIR}"/se/SRR12875558_1.fastq.gz \
+-L \
+-a \
+| samtools sort \
+-@ 47 \
```
Comment (sofstam) [inline assets/ci/README.md]: Same as above.

**Review by sofstam — COMMENTED** [review]
Looks good, a few comments only!

File: assets/ci/README.md (hunk 1 above)
Comment (ljmesi) [reply assets/ci/README.md]: Thanks for the reviews! It's correct now.

File: assets/ci/README.md (hunk 2 above)
Comment (ljmesi) [reply assets/ci/README.md]: Yes. I reduced the number of threads to 10 and 16 now.

File: assets/ci/README.md (hunk 3 above)
Comment (ljmesi) [reply assets/ci/README.md]: Fixed now.

File: assets/ci/README.md (hunk 4 above)
Comment (ljmesi) [reply assets/ci/README.md]: Fixed now.

File: assets/ci/README.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+-title "Ungulate tetraparvirus 3 db" \
+-taxid "${TAXID}" \
+-out "${BLAST_DB_PATH}" \
+-logfile makeblastdb_"${TAXID}".log
+```
+
+Test retrieving taxid 1511916:
+
+```bash
+CURRENT_DIR=/home/lauri/Desktop/gmsmetapost_dev/our_fork/gmsmetapost/temp/20220907_ci_testing
```
Comment (sofstam) [inline assets/ci/README.md]: Shall this pathway point to general gmsmetapost repo?

File: assets/ci/README.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+```
+
+## Obtaining of suitable minimal fastq files
+
+### Map all reads to *Ungulate tetraparvovirus 3* and filter out unmapped reads
+
+#### Paired end reads
+
+```bash
+FASTQ_FILES_DIR=/home/lauri/Desktop/gmsmetapost_dev/our_fork/gmsmetapost/assets/input/
```
Comment (sofstam) [inline assets/ci/README.md]: Same here?

**Review by sofstam — APPROVED** [review]
Looks great! Just a minor comment on some pathways.

File: assets/ci/README.md (hunk 5 above)
Comment (ljmesi) [reply assets/ci/README.md]: No, it's actually quite unnecessary to have it. I removed it now as well as the unneeded `cd` commands. They would have just made sure that the user is in the correct directory every time.

File: assets/ci/README.md (hunk 6 above)
Comment (ljmesi) [reply assets/ci/README.md]: I changed this to relative path (so that gmsmetapost is the current working directory) so it should be okay now.

**Review by sofstam — APPROVED** [review]
Ready for merging :)

**Pull request merged**